modules/indexing/src/test/java/org/apache/ignite/internal/processors/query/stat/StatisticsConfigurationTest.java

```diff
@@ -0,0 +1,522 @@
+            "PUBLIC", "SMALL", (s -> assertNull(s.data().get("A"))));
+        checkStatisticsInMetastore(grid(2).context().cache().context().database(), TIMEOUT,
+            "PUBLIC", "SMALL", (s -> assertNull(s.data().get("A"))));
+
+        startGrid(1);
+
+        checkStatisticsInMetastore(grid(1).context().cache().context().database(), TIMEOUT,
+            "PUBLIC", "SMALL", (s -> assertNull(s.data().get("A"))));
+
+        waitForStats("PUBLIC", "SMALL", TIMEOUT, checkTotalRows,
+            (stats) -> {
+                stats.forEach(s -> assertNull("Invalid stats: " + stats, s.columnStatistics("A")));
+            }
+        );
+    }
+
+    /**
+     * Check drop statistics when table is dropped.
+     */
+    @Test
+    public void dropTable() throws Exception {
+        startGrids(3);
+
+        grid(0).cluster().state(ClusterState.ACTIVE);
+
+        createSmallTable(null);
+        createSmallTable("_A");
+
+        updateStatistics(
+            new StatisticsTarget("PUBLIC", "SMALL"),
+            new StatisticsTarget("PUBLIC", "SMALL_A"));
+
+        waitForStats("PUBLIC", "SMALL", TIMEOUT, checkTotalRows, checkColumStats);
+        waitForStats("PUBLIC", "SMALL_A", TIMEOUT, checkTotalRows, checkColumStats);
+
+        dropSmallTable(null);
+
+        waitForStats("PUBLIC", "SMALL", TIMEOUT,
+            (stats) -> stats.forEach(s -> assertNull(s)));
+
+        waitForStats("PUBLIC", "SMALL_A", TIMEOUT, checkTotalRows, checkColumStats);
+
+        for (Ignite ign : G.allGrids()) {
+            checkStatisticsInMetastore(((IgniteEx)ign).context().cache().context().database(), TIMEOUT,
+                "PUBLIC", "SMALL", (s -> assertNull(s)));
+        }
+    }
+
+    /**
+     * Check drop statistics when table's column is dropped.
+     */
+    @Test
+    public void dropColumn() throws Exception {
+        startGrids(3);
+
+        grid(0).cluster().state(ClusterState.ACTIVE);
+
+        createSmallTable(null);
+
+        updateStatistics(new StatisticsTarget("PUBLIC", "SMALL"));
+
+        waitForStats("PUBLIC", "SMALL", TIMEOUT, checkTotalRows, checkColumStats);
+
+        sql("DROP INDEX SMALL_B");
+        sql("ALTER TABLE SMALL DROP COLUMN B");
+
+        waitForStats("PUBLIC", "SMALL", TIMEOUT,
+            (stats) -> stats.forEach(s -> {
+                assertNotNull(s.columnStatistics("A"));
+                assertNotNull(s.columnStatistics("C"));
+                assertNull(s.columnStatistics("B"));
+            }));
+
+        for (Ignite ign : G.allGrids()) {
+            checkStatisticsInMetastore(((IgniteEx)ign).context().cache().context().database(), TIMEOUT,
+                "PUBLIC", "SMALL", (s -> assertNull(s.data().get("B"))));
+        }
+    }
+
+    /**
+     * Check drop statistics when table's column is dropped and node with old statistics joins to cluster
+     * after drop the column.
+     */
+    @Test
+    public void dropColumnWhileNodeDown() throws Exception {
+        if (persist)
```

Comment:
Maybe if ( **!** persist)?

Reply:
The persist node must be cleaned in case the cache configuration is changed when the node was down.